appnexus/client.py

```diff
@@ -113,33 +110,32 @@ def update_token(self):
         if None in self.credentials.values():
             raise RuntimeError("You must provide an username and a password")
         credentials = dict(auth=self.credentials)
-        url = self.test_url if self.test else self.url
-        response = requests.post(url + "auth",
-                                 json=credentials)
-        data = response.json()["response"]
-        if "error_id" in data and data["error_id"] == "NOAUTH":
+        response = self.create("auth", credentials)
+        if "error_id" in response and response["error_id"] == "NOAUTH":
             raise BadCredentials()
-        if "error_code" in data and data["error_code"] == "RATE_EXCEEDED":
-            time.sleep(150)
-            return
-        if "error_code" in data or "error_id" in data:
-            raise AppNexusException(response)
-        self.token = data["token"]
+        self.token = response["token"]
         self.save_token()
         return self.token
 
     def check_errors(self, response, data):
         """Check for errors and raise an appropriate error if needed"""
         if "error_id" in data:
-            error_id = data["error_id"]
-            if error_id in self.error_ids:
-                raise self.error_ids[error_id](response)
-        if "error_code" in data:
-            error_code = data["error_code"]
-            if error_code in self.error_codes:
-                raise self.error_codes[error_code](response)
-        if "error_code" in data or "error_id" in data:
-            raise AppNexusException(response)
+            raise type(data["error_id"], (AppNexusException,), {})(response)
```

Comment:
No need to create a new type if we catch it right after. Let's just check the "error_id" directly in an `handle_errors` method.